### PR TITLE
ss-327: add ability to create transaction

### PR DESCRIPTION
**Pull request by Gandrejko**

### CODE REVIEW

**Review by anton-otroshchenko — DISMISSED** [review]

**Comment by Evcat6** [comment]
Conflicts

**Review by anton-otroshchenko — APPROVED** [review]

**Comment by teslmik** [comment]
chrome_a3XtuLaZm2

File: frontend/src/bundles/common/components/transaction-modal/transaction-image.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const inputReference = useRef<HTMLInputElement>(null);
+
+    const handleClick = useCallback((): void => {
+        if (inputReference.current) {
+            inputReference.current.click();
+        }
+    }, []);
+
+    return (
+        <button
```
Comment (Evcat6) [inline frontend/src/bundles/common/components/transaction-modal/transaction-image.tsx:24]: Can we use there our common components?

File: frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx [hunk 2]
```diff
@@ -100,10 +100,10 @@ const BudgetDetails = (): JSX.Element => {
     }, [dispatch]);
 
     useEffect(() => {
-        const spentResult = transactions.reduce(
-            (accumulator, current) => accumulator + current.amount,
-            0,
-        );
+        let spentResult = 0;
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx:103]: i think this should be helper

Comment (Evcat6) [reply frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx:103]: maybe `getSpent` or something like this

**Review by Evcat6 — CHANGES_REQUESTED** [review]

**Review by Evcat6 — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Pull request merged**